packages/salesforcedx-vscode-core/src/orgBrowser/metadataOutlineProvider.ts

```diff
@@ -24,33 +37,144 @@ export class MetadataOutlineProvider
     this.defaultOrg = defaultOrg;
   }
 
-  public async refresh(defaultOrg?: string): Promise<void> {
-    if (!isNullOrUndefined(defaultOrg)) {
-      this.defaultOrg = defaultOrg;
+  public async onViewChange() {
+    if (await this.getDefaultUsernameOrAlias()) {
+      this.internalOnDidChangeTreeData.fire();
     }
+  }
+
+  public async refresh(defaultOrg?: string): Promise<void> {
+    await this.getDefaultUsernameOrAlias();
     this.internalOnDidChangeTreeData.fire();
   }
 
   public getTreeItem(element: BrowserNode): vscode.TreeItem {
     return element;
   }
 
-  public getChildren(element?: BrowserNode): Promise<BrowserNode[]> {
+  public async getChildren(element?: BrowserNode): Promise<BrowserNode[]> {
     if (isNullOrUndefined(element)) {
       if (!isNullOrUndefined(this.defaultOrg)) {
         const org = new BrowserNode(this.defaultOrg, NodeType.Org);
         return Promise.resolve([org]);
       } else {
-        return Promise.resolve([]);
+        const emptyDefault = new BrowserNode(
+          'No default org set',
+          NodeType.EmptyNode
+        );
+        return Promise.resolve([emptyDefault]);
+      }
+    } else if (element.type === NodeType.Org) {
+      const metadataTypes = await this.getTypes();
+      element.children = metadataTypes;
+      return Promise.resolve(element.children);
+    } else if (element.type === NodeType.MetadataType) {
+      const metadataCmps = await this.getComponents(element);
+      if (isNullOrUndefined(metadataCmps) || metadataCmps.length === 0) {
+        const emptyDefault = new BrowserNode(
+          'No components available',
+          NodeType.EmptyNode
+        );
+        element.children = [emptyDefault];
+      } else {
+        element.children = metadataCmps;
       }
-    } else {
       return Promise.resolve(element.children);
     }
+
+    return Promise.resolve([]);
   }
 
-  public async getDefaultUsernameOrAlias() {
+  public async getTypes(): Promise<BrowserNode[]> {
+    const username = this.defaultOrg!;
+    const typesFolder = await getTypesFolder(username);
+    const typesPath = path.join(typesFolder, 'metadataTypes.json');
+
+    let typesList: string[];
+    if (!fs.existsSync(typesPath)) {
+      try {
+        const result = await forceDescribeMetadata(typesFolder);
+        typesList = buildTypesList(result, undefined);
+      } catch (e) {
+        const errorNode = new BrowserNode(parseErrors(e), NodeType.EmptyNode);
+        return [errorNode];
+      }
+    } else {
+      typesList = buildTypesList(undefined, typesPath);
+    }
+    const nodeList = [];
+    for (const type of typesList) {
+      const typeNode = new BrowserNode(type, NodeType.MetadataType);
+      nodeList.push(typeNode);
+    }
+    return nodeList;
+  }
+
+  public async getComponents(
+    metadataType: BrowserNode
+  ): Promise<BrowserNode[]> {
+    const username = this.defaultOrg!;
+    const componentsPath = await getComponentsPath(
```

Comment:
I'm wondering if metadataCmp.ts can be responsible for doing more of this work. The provider is really only concerned with having a list of the component names to construct the nodes and shouldn't care whether it comes from the cache or remote.